### PR TITLE
Contextvars

### PR DESCRIPTION
**Pull request by bloodearnest**
This includes:
 - dropping 3.4 support and adding 3.7
 - asyncio support for context
 - 3.5/3.6 only dependencies on contextvars and aiocontextvars
 - py2 minimal polyfill for contextvars
 - reworking tests
 - fix limbo script to include environment markers

### CODE REVIEW

File: requirements.tests.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ pytest==4.4.1
 freezegun==0.3.11
 pytest-cov==2.6.1
 pytest-postgresql==1.4.0
+mirakuru==1.1
```
Comment (bloodearnest) [inline requirements.tests.txt]: Mirakuru was an unpinned dependency of pytest-postgresql, and version 2 drops py2 support, so pin it

**Comment by bloodearnest** [comment]
This pull request **introduces 1 alert** when merging 2c817f466965159a29689a3df6056dc203217720 into fbdf0f79b0eba2a5a27b8330f6ef9b72c0c38596 - [view on LGTM.com](https://lgtm.com/projects/g/canonical-ols/talisker/rev/pr-9f2f7b4fbad8a295f1f7ab7eb478b3ad3138ec2f)

**new alerts:**

* 1 for Unused import

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: talisker/context.py [hunk 2]
```diff
@@ -30,27 +30,95 @@
...
+    import contextvars
+
+    # enable asyncio aware contextvars in 3.5.3+/3.6
+    if pkg_is_installed('aiocontextvars'):
+        import asyncio
+        # aiocontextvars only supports python 3.5.3+
+        if hasattr(asyncio, '_get_running_loop'):
+            import aiocontextvars  # NOQA
+        else:
+            # TODO: warn user?
```
Comment (tonysimpson) [inline talisker/context.py]: Can you add a comment to explain the impact of this.

**Comment by bloodearnest** [comment]
This pull request **introduces 1 alert** when merging 7b5abd7f4045ab9796f512c96895206af3e84c6e into fbdf0f79b0eba2a5a27b8330f6ef9b72c0c38596 - [view on LGTM.com](https://lgtm.com/projects/g/canonical-ols/talisker/rev/pr-d4cf6b1628d1db811f10adf951cdb4e47904dc19)

**new alerts:**

* 1 for Unused import

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**